### DISCUSSION TITLE
FIP0001 - specs update process

### DISCUSSION THREAD

**Discussion by dd45e640b42e6da7da96faee3996ef7c** [filecoin-project/FIPs#678] [Enhancements - Organizational]
[FIP0001 states](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md#fip-rationale):

> However, we also **require** that, where appropriate, Technical FIPs are also reflected in the protocol's specification, which is the primary source of reference for other developers. The [Filecoin Spec site](https://spec.filecoin.io/) will in turn point back to the FIPs that have changed some part of the protocol, as well as the FIP's revision history.

currently the last commit to the `master` branch of the [specs repo](https://github.com/filecoin-project/specs) was made on May 25th 2022! 

How is this supposed to be handled?  There seem to be a lot of FIPs not represented in the specs at the current point in time.

Seeking clarification here about at what point  the specs need to be updated to represent the implementation of a FIP.

**Comment by raulk** [comment]
Side note that spec reconciliation is also a challenge in other protocols like Ethereum (which move slower than Filecoin). The current published yellow paper captures Berlin and is 6 hard forks behind Paris (currently live).

https://ethereum.github.io/yellowpaper/paper.pdf
https://ethereum.org/en/history/

Solving this efficiently will IMO require a dedicated spec maintenance squad that works in the background. Synchronously throttling the FIP innovation process on FIP authors updating the spec (esp. in the state it's in today) is shooting ourselves in the foot.

**Reply by jsoares** [reply]
I disagree. I think recovering from the current situation definitely requires a spec squad -- and I don't know that we'll ever get one, so your conclusion might still be right. However, ignoring that small detail, this requirement shouldn't be an issue in steady state. Morever, not having a spec likely slows us down more than having to update it, and making changes without properly specifying them is... less than ideal.

**Comment by raulk** [comment]
Let's remove this requirement from FIP-0001 since it was DOA, and start a thread on spec update/maintenance? This can be crowdsourced IMO -- a good model is Ethereum Cat Herders.
